client/app/utils/i18n.js

```diff
@@ -0,0 +1,39 @@
+// This file has two tasks:
+//
+// 1. Load the language bundle if we are in node environment
+// 2. Require the i18n-js library and export it
+//
+// The language bundle is not loaded if we are in the browser
+// environment because there we get the bundle from a separate
+// long cached script-tag
+//
+// The language bundle is loaded in a global I18n variable like this:
+//
+// `global.I18n.translations["en"] = <translation json>`
+//
+// The i18n-js library is able to read the language bundle from that global
+//
+
+import { bind } from 'lodash';
+
+if (typeof window === 'undefined') {
+  try {
+    require('../../../app/assets/javascripts/i18n/all.js'); // eslint-disable-line no-undef
+  } catch (e) {
+    console.warn("Can't load language bundle all.js"); // eslint-disable-line no-console
+  }
+}
+
+// It seems that we need to use `require` here instead of `import`.
+// With `import` the translations seem not to be populated.
+const I18n = require('i18n-js'); // eslint-disable-line no-undef
```

Comment:
'Require' loadings are synchronous, maintaining the order. However, I don't understand why this is the order needed - I thought it was supposed to be the other way around. Btw. have you tried loading the page with throttling connection?